crates/secrets/src/aws_secrets_manager.rs

```diff
@@ -32,12 +50,99 @@ impl AwsSecretsManager {
     pub fn new() -> Self {
         Self {}
     }
+
+    /// Initializes AWS configuration and verifies AWS credentials.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error:
+    /// - If the AWS configuration cannot be loaded.
+    /// - If the call to STS `get_caller_identity` fails, which might be due to invalid or expired AWS credentials.
+    pub async fn init(self) -> Result<(), Box<dyn std::error::Error>> {
+        let config = aws_config::defaults(BehaviorVersion::v2023_11_09())
+            .load()
+            .await;
+
+        let sts_client = aws_sdk_sts::Client::new(&config);
+
+        if sts_client.get_caller_identity().send().await.is_err() {
+            return Err(Box::new(Error::InvalidAwsCredentials {}));
+        }
+
+        Ok(())
+    }
 }
 
 #[async_trait]
 impl SecretStore for AwsSecretsManager {
     #[must_use]
-    async fn get_secret(&self, _secret_name: &str) -> Option<Secret> {
+    async fn get_secret(&self, secret_name: &str) -> Option<Secret> {
```

Comment:
We should revisit this signature - I'm not a fan of swallowing errors and returning `None` - as that doesn't allow the caller to distinguish between whether a secret isn't there, or there was some error getting the secret.

Reply:
We can log it as a bug

Reply:
Reported as https://github.com/spiceai/spiceai/issues/1070